### PR TITLE
fix(render-ui): create function that parses html in order to fix PD-771

### PR DESCRIPTION
**Pull request by PatriciaRomaniuc**
https://illuminate.atlassian.net/browse/PD-771

### CODE REVIEW

**Review by andreeapescar — APPROVED** [review]

File: packages/demo/pages/editable-html.js [hunk 1]
```diff
@@ -110,6 +113,12 @@ class RteDemo extends React.Component {
     this.setState({ markup });
   };
 
+  onChangeMarkupText = markupText => {
+    log('onChangeMarkupText: ');
+    this.setState({ markupText });
```
Comment (edeustace) [inline packages/demo/pages/editable-html.js:118]: why 2 calls to `setState`? why not just 1?

File: packages/demo/pages/editable-html.js [hunk 2]
```diff
@@ -101,7 +102,9 @@ class RteDemo extends React.Component {
       disabled: false,
       width: '',
       height: '',
-      keypadMode: 'everything'
+      keypadMode: 'everything',
+      markupText: html,
+      htmlHasText: true
```
Comment (edeustace) [inline packages/demo/pages/editable-html.js:107]: this name isn't clear to me - why not just `hasText`? The client comp doesn't care if it's html or not?

File: packages/render-ui/src/__tests__/parseHtmlHasText.test.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import { parseHtmlHasText } from '../parseHtmlHasText';
```
Comment (edeustace) [inline packages/render-ui/src/__tests__/parseHtmlHasText.test.js:3]: `hasText`? or `hasContent` - you're looking for a boolean so typically you have a `is*` or `has*` function name? also the file name I think we use `kebab-case` if that's the case - make it consistent

File: packages/render-ui/src/parseHtmlHasText.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+export const parseHtmlHasText = input => {
```
Comment (edeustace) [inline packages/render-ui/src/parseHtmlHasText.js:1]: rename please - see above ..
also I don't see why you are doing things like this  could you not simplify like: 
```javascript
const dp = new DOMParser();

let markupToText = (s) => { 
   const root = dp.parseFromString(s, 'text/html');
   return root.body.textContent;
}

let hasContent = (s) => {
  if(!s){
    return false;
  }
  let tc = markupToText(s);
  return tc && tc.trim() ? true : false;
}
console.log('1', true, hasContent('<div>Rationale</div>'));
console.log(2, true, hasContent('Rationale'));
console.log(3, false, hasContent('<div></div>'));
console.log(4, false, hasContent('<div> </div>'));
console.log(5, false, hasContent(' '));
console.log(6, false, hasContent(null));
```

?

**Review by edeustace — CHANGES_REQUESTED** [review]

File: packages/demo/pages/editable-html.js (hunk 2 above)
Comment (PatriciaRomaniuc) [reply packages/demo/pages/editable-html.js:107]: i'll change it

File: packages/render-ui/src/parseHtmlHasText.js (hunk 4 above)
Comment (PatriciaRomaniuc) [reply packages/render-ui/src/parseHtmlHasText.js:1]: sure, I'll simplify the function

**Pull request merged**